src/lib/lessons.ts

```diff
@@ -64,8 +64,10 @@ const lessonQuery = groq`
       title,
       'thumb_url': thumbnailUrl,
       resource->_type == 'videoResource' => {
-        'media_url': resource->hlsUrl,
-        duration,
+        ...(resource-> {
+          'media_url': hlsUrl,
+          duration
+        })
```

Comment:
What's up with `...(resource-> { })`?

This GROQ trick allows you to grab the `resource` once and access
multiple values off of that. The way it was before, I believe under the
hood it does the equivalent of a subquery to get each ref attribute.